src/docfx/build/toc/TocMap.cs

```diff
@@ -179,13 +185,29 @@ private static (int subDirectoryCount, int parentDirectoryCount) GetRelativeDire
         var allTocs = new List<(FilePath file, HashSet<FilePath> docs, HashSet<FilePath> tocs, bool shouldBuildFile)>();
         var includedTocs = new HashSet<FilePath>();
         var allServicePages = new List<FilePath>();
+        var tocFilesFromBuildScope = _buildScope.GetFiles(ContentType.Toc);
+        var (originalReferenceTOCs, targetReferenceTOCs) = GetOriginalReferenceTocWithTargetReferenceToc(tocFilesFromBuildScope);
 
         // Parse and split TOC
         ParallelUtility.ForEach(
             scope,
             _errors,
-            _buildScope.GetFiles(ContentType.Toc),
-            file => SplitToc(file, _tocParser.Parse(file, _errors), allTocFiles));
+            tocFilesFromBuildScope.Concat(targetReferenceTOCs),
```

Comment:
nit: `tocFilesFromBuildScope.Concat(targetReferenceTOCs).Except(originalReferenceTOCs)` so that you don't need line 198

Reply:
done